Techtonica Mod Loader/MainWindow.xaml.cs

```diff
@@ -1,152 +1,224 @@
-        {
-            CallUpdateWindow();
-        }
-
-        private void Button_Launch_Vanilla_Click(object sender, RoutedEventArgs e)
-        {
-            ProcessStartInfo start = new ProcessStartInfo();
-            start.Arguments = e.ToString();
-            start.FileName = "Techtonica.exe";
-            start.WorkingDirectory = GameLocation;
-            start.UseShellExecute = true;
-            SendDebugLine("Launching from directory: "+ start.WorkingDirectory);
-            SendDebugLine("Target: " + start.FileName);
-            start.WindowStyle = ProcessWindowStyle.Normal;
-            start.CreateNoWindow = false;
-            int exitCode;
-            SendDebugLine("Launching game in Vanilla mode!");
-            try { 
-                using (Process proc = Process.Start(start))
-                {
-                    proc.WaitForExit();
-
-                    // Retrieve the app's exit code
-                    exitCode = proc.ExitCode;
-                }
-            }
-            catch (Win32Exception ex)
-            {
-
-                SendDebugLine(ex.Message);
-                Button_Launch_Vanilla.IsEnabled = true;
-                Button_Launch_Vanilla.Content = "Game not found!";
-                Button_Launch_Vanilla.Background = LaunchVanillaDisabledBrush;
-            }
-        }
-    }
-}
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+using AutoUpdaterDotNET;
+using System.Diagnostics;
+using Microsoft.Win32;
+using System.ComponentModel;
+using System.Reflection;
+using System.Net;
+using System.IO;
+using Techtonica_Mod_Loader.Classes.Globals;
+using Techtonica_Mod_Loader.Classes;
+using Techtonica_Mod_Loader.Panels;
+
+namespace Techtonica_Mod_Loader
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        public MainWindow()
+        {
+            InitializeComponent();
+        }
+
+        // Objects & Variables
+
+        public static MainWindow current => (MainWindow)Application.Current.MainWindow;
+
+        // ToDo: Move To ProgramData
+        public string GameLocation; 
+        public Brush LaunchVanillaDisabledBrush = new SolidColorBrush(Colors.DarkRed);
+        public string version = FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location).FileVersion;
+        public string SelfLoc = Directory.GetParent(AppContext.BaseDirectory).FullName;
+        //public UpdateInfoEventArgs UpdateArgs = new UpdateInfoEventArgs();
+
+        // Events
+
+        private void OnProgramLoaded(object sender, RoutedEventArgs e) {

```

Comment:
Simple function can do?
Don't see a need for init data that is run once to be pushed to an event.
This will bloat the data...